persistence/remote/src/test/java/org/infinispan/persistence/remote/RemoteStoreConfigTest.java

```diff
@@ -35,6 +36,7 @@
    public void startUp() {
       cacheManager = TestCacheManagerFactory.createCacheManager(hotRodCacheConfiguration());
       assertEquals(cacheManager.getCache().size(), 0);
+      TestingUtil.registerTranscoders(cacheManager, new ProtostreamBinaryTranscoder());
```

Comment:
I'm surprised the server doesn't already do this.

Reply:
It's normally added by `remote-query-server`'s LifecycleManager, however that's not called when creating the HotrodServer via `HotRodTestingUtil.startHotRodServer`

Reply:
I'm still surprised... `HotRodTestingUtil.startHotRodServer` doesn't do anything to skip the initialization of certain modules.

Reply:
@danberindei Neither the remote store module, or  `infinispan-server-hotrod` has a dependency on `remote-query-server`, therefore it's not possible for the `RemoteStoreConfigTest` to have the query server's LifecycyleManager executed.

Reply:
`persistence-remote` could add `remote-query-server` as a test dependency. 

But query isn't really needed to support the protobuf media type on the server, so `ProtostreamBinaryTranscoder` and the other transcoders in the same package could and IMO should be registered by `org.infinispan.server.core.LifecycleCallbacks`.

Reply:
I've moved the `ProtostreamBinaryTranscoder` to server-core, however it does not make sense for the other transcoders to be moved as they rely on the `ProtobufMetadataManager`'s SerializationContext.